dali/python/nvidia/dali/pipeline.py

```diff
@@ -51,10 +51,15 @@ class Pipeline(object):
 Parameters
 ----------
 `batch_size` : int, optional, default = -1
-    Batch size of the pipeline. Negative values for this parameter
+    Maximum batch size of the pipeline. Negative values for this parameter
```

Comment:
Again, does this need to be called maximum batch size? This will encourage  users to set it to some super high values and it's especially misleading when the readers still don't support variable batch size and will always read the maximal batch size. 'Default batch size' would be better here, imo.

Reply:
As above, setting `max_batch_size` allows for preallocations, which wouldn't be possible with `default_batch_size`.

Additionally, with `max_batch_size` we'll be able to catch some part of OOM errors, which in case of `default` one would emerge only after particular amount of training epochs. I guess user would appreciate knowing this beforehand ;)